server/broadcast_service.py

```diff
@@ -0,0 +1,120 @@
+from aio_pika import DeliveryMode
+
+from .config import config
+from .core import Service
+from .decorators import with_logger
+from .game_service import GameService
+from .games import GameState
+from .message_queue_service import MessageQueueService
+from .player_service import PlayerService
+from .timing import LazyIntervalTimer
+
+
+@with_logger
+class BroadcastService(Service):
+    """
+    Broadcast updates about changed entities.
+    """
+
+    def __init__(
+        self,
+        server: "ServerInstance",
+        message_queue_service: MessageQueueService,
+        game_service: GameService,
+        player_service: PlayerService,
+    ):
+        self.server = server
+        self.message_queue_service = message_queue_service
+        self.game_service = game_service
+        self.player_service = player_service
+
+    async def initialize(self):
+        await self.message_queue_service.declare_exchange(
+            config.MQ_EXCHANGE_NAME
+        )
+
+        # Using a lazy interval timer so that the intervals can be changed
+        # without restarting the server.
+        self._broadcast_dirties_timer = LazyIntervalTimer(
+            lambda: config.DIRTY_REPORT_INTERVAL,
+            self.report_dirties,
+            start=True
+        )
+        self._broadcast_ping_timer = LazyIntervalTimer(
+            lambda: config.PING_INTERVAL,
+            self.broadcast_ping,
+            start=True
+        )
+        self._logger.debug("Broadcast service initialized")
+
+    async def report_dirties(self):
```

Comment:
As someone new to the code base I'd love a brief docstring here.

Reply:
How about that?